### PR TITLE
add a progress_status_timeout option for allow_upload

### PR DESCRIPTION
**Pull request by iwarshak**
This adds an option to `allow_upload` which will let you change the timeout wait (or eliminate the timeout completely) for progress messages to be acknowledged by the LiveView. 

I am running this code because of an [issue](https://github.com/phoenixframework/phoenix_live_view/issues/1394) I was experiencing in production where a slow response to progress messages would cause the JS to completely reload the page and lose the file that was already uploaded.

_(I understand that my situation may be an outlier or that there may be a better solution to this problem but I wanted to offer this potential contribution up anyway.)_

### CODE REVIEW

**Comment by chrismccord** [comment]
Thanks for the PR Ian! I think the real culprit of your issues is the LV performing blocking work in another callback under load and queue the progress events. Fixing that issue is the best option, but I do think you've uncovered some optimization work we could do to help avoid the LV going into further overload on progress events. On the client, we could wait until the progress event is acknowledged until sending the next chunk, but this will potentially slow down uploads. Another option is only sending the next progress event after we've received an ack from the previous, thereby avoiding extra messages. Lastly, a third option is to support a `chunk_size: :auto` where we compute a reasonable chunk size based on the file size so we are sending fewer progress updates, or sending fewer progress for larger file sizes. If you'd like to explore these in a new PR, let me know! Thanks!

**Comment by iwarshak** [comment]
@chrismccord What do you think of a 4th option, which would be: never sending more than 100 progress events (1 event percentage point of progress)? 

As it currently stands, a 40 meg file with the default upload chunk size (64kb) would send ~625 progress events. If the end user has a super fast internet connection, those messages could pile up fairly rapidly.

**Comment by chrismccord** [comment]
> or sending fewer progress for larger file sizes

That's similar to what I was thinking for # 4. Sounds good to me! Do you want to send a PR?

**Comment by iwarshak** [comment]
Sure - I can give it a shot. So you are *sure* that a delayed or drop progress event is enough to blow up a whole upload?

**Comment by chrismccord** [comment]
> So you are _sure_ that a delayed or drop progress event is enough to blow up a whole upload

Can you rephrase? We require a `100` progress event to indicate the completion of the upload. The previous progress events awaiting an ack from the previous event is fine, if that's what you're asking

**Comment by iwarshak** [comment]
Here's my second attempt: https://github.com/phoenixframework/phoenix_live_view/pull/1461
